### DISCUSSION TITLE
Loading .60 ui on runtime

### DISCUSSION THREAD

**Discussion by saona-raimundo** [slint-ui/slint#811] [Q&A] [answered]
I am sorry if this topic is repeated, but is there a way to load .60 ui on runtime?

I am a Rust programmer, read the documentation and it seems that either the ui has to be hard-coded within the macro `sixtyfps!` or loaded at compile time from a `.60` file.

## Use-case

In general, a model like Server-side rendering, where an external entity gives you a compiled ui to render.
In my particular case, I wanted to use sixtyFPS as the ui backend of another framework and hit this wall.

## Particular example

Say that out of an http request, I get a `.60` file and would like to show it to the user. What would you do?

Thanks beforehand!

**Comment by tronical** [comment, marked as answer]
Yes, we have the interpreter crate for exactly this use case: https://docs.rs/sixtyfps-interpreter/0.1.5/sixtyfps_interpreter/

We fail to link to it from the main page and docs - I’ll try to fix that. @ogoffart or did I miss something and we didn’t want to make it public yet?

**Reply by saona-raimundo** [reply]
Thank you! I did not see that crate.

> We fail to link to it from the main page and docs - I’ll try to fix that.

That would be great!

> did I miss something and we didn’t want to make it public yet?

The crate [has been in crates.io](https://crates.io/crates/sixtyfps-interpreter/versions) since a year ago, so it should be fine(?)

**Reply by tronical** [reply]
Yes, it’s fine to use it and we’re maintaining it like public api. :)
